[ADD] Attendances: Reporting (#13332)
odoo/documentation: content/applications/hr/attendances/attendance_reporting.rst

====================
Attendance reporting
====================

The **Attendances** app report helps managers find attendance issues with employees before they become
a problem. The report can help determine trends by determining :ref:`which employees are working more
overtime <attendances/overtime>`, and who is :ref:`not working their full shifts <attendances/absent>`.

View report
===========

To view the attendance report, click :guilabel:`Reporting` in the top menu. The default report displays

> attendance information for current employees over the past 3 months, in a default :icon:`oi-view-pivot`

Reviewer: according to our [style guide](https://docs.google.com/document/d/1H0FDqprekFG7xei7vbDslOxvRJQCPaaMY4ykXj28ejw/edit?tab=t.0#heading=h.lhdotfd6vosf), we spell out numbers smaller than 10 (I had to look it up)
```suggestion
attendance information for current employees over the past three months, in a default :icon:`oi-view-pivot`
```

.. image:: reporting/reporting.png
   :alt: The default report view, with all the optional view buttons highlighted.

.. _attendances/overtime:

Use case: monthly overtime
==========================

Overtime affects a company's bottom line, as most companies budget for employee salaries. When salaries exceed
the expected amount, companies may see a rise in overhead costs when salary budgets are exceeded. This is
why the attendance report is needed, to ensure no excessive overtime is logged.

To view a specific month's overtime, by employee, click :guilabel:`Reporting` in the top menu of the
**Attendances** app. Remove the default :icon:`oi-group` :guilabel:`Date: Month > Employee` grouping in the
search bar. Click the :icon:`fa-caret-down` :guilabel:`(Toggle Search Panel)` icon at the end of the search bar,
then click :icon:`fa-caret-down` :guilabel:`Date`, and click the desired month.

Next, click the :guilabel:`Measures` :icon:`fa-caret-down` button in the upper-left corner, and deselect
:guilabel:`Expected Hours` and :guilabel:`Worked Hours`. This presents the :guilabel:`Difference` and :guilabel:`Balance`
attendance data for the selected month, grouped by employee.

> The :guilabel:`Difference` is the total logged overtime for the month, and the :guilabel:`Balance` is the unapproved
> overtime. Click :guilabel:`Difference` to sort by the amount of overtime. The highest positive number is the most

Reviewer: ```suggestion
:guilabel:`Difference` shows the month's total overtime logged, and the :guilabel:`Balance` shows the portion still unapproved. Click :guilabel:`Difference` to sort by the amount of overtime. The highest positive number is the most
```

overtime logged for that month.

.. example::
   In the below example, thirty-eight hours and fifty-three minutes of overtime was approved for the month of April.

>    :guilabel:`Michael Williams` has the most overtime, with ten hours and twenty-five minutes logged. Of that overtime,
>    only twenty-five minutes was unapproved, for a total of ten paid overtime hours.

Reviewer: ```suggestion
   :guilabel:`Michael Williams` has the most overtime, with 10 hours and 25 minutes logged. Of that overtime,
   only 25 minutes was unapproved, for a total of 10 paid overtime hours.
```

Had to look up our style guide for numbers :rofl: 

![image](https://github.com/user-attachments/assets/1b6dd699-e2cd-45cf-b450-4032a492f600)

   .. image:: reporting/overtime.png
      :alt: The attendances report, showing only overtime for the month of April.

.. _attendances/absent:

Use case: absenteeism
=====================

Employees with a negative :guilabel:`Difference` indicate that they have not worked their expected shifts. This can
be from forgetting to log in and out of work, or for working less hours than expected, without any approved time
off.

To view the attendance data for the last three months, to identify any trends, first open the **Attendances**

> app report. Next, click the :guilabel:`Measures` :icon:`fa-caret-down` button in the upper-left corner, and deselect
> :guilabel:`Balance`, :guilabel:`Expected Hours`, and :guilabel:`Worked Hours`.

Reviewer: To minimize double negatives ("deselect", and the listed fields are NOT what we should pick), let's make it an instructions

```suggestion
app report. Next, click the :guilabel:`Measures` :icon:`fa-caret-down` button in the upper-left corner, and ensure **only** 
:guilabel:`Difference` is selected.
```

Author: That's a WAY better way to say that!

> Next, click :icon:`fa-minus-square-o` :guilabel:`Total` at the top of the rows, then click :icon:`fa-plus-square`

Reviewer: I was a bit confused about why we clicked minus then plus on the same button, so hopefully this adds some context (:
```suggestion
Next, reset the row data by clicking :icon:`fa-minus-square-o` :guilabel:`Total` at the top of the rows. Then click :icon:`fa-plus-square`
```

> :guilabel:`Total`, then :guilabel:`Employee`. Click :guilabel:`Difference` in the last column, to sort by the total

Reviewer: ```suggestion
:guilabel:`Total` and select :guilabel:`Employee`. Click the :guilabel:`Difference` column name to sort by the total
```

difference in logged hours from the expected hours.

The employee who worked the least has the highest negative balance. Click on any of the numbers for the employee
to view the detailed individual attendance records.

.. example::
   In the below example, the company has only been opened for two months, so data is only available for the last two

>    months, even though the filter :guilabel:`Last 3 Months` is active.

Reviewer: is it necessary to mention that the company was only open for 2 months? I personally didn't think it was weird to only see two columns (:

Author: I just thought I needed to explain why it has a 3-month filter but only shows 2 months of data.

   From this report, it is determined that :guilabel:`Abigail Peterson` worked the least amount of hours over

>    the past two months, working twenty-five hours less than was expected.

Reviewer: ```suggestion
   the past two months, working 25 hours less than expected.
```
